.github/workflows/publish-rust.yml

```diff
@@ -130,6 +130,9 @@ jobs:
           git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
           git config --global user.name "github-actions[bot]"
 
+      - name: Rebase (in case any changes landed after)
```

Comment:
I think we need to do a release build first because that's what took so long

Reply:
I just meant that the publish takes a long time due to the release build it does.. so it doesn't really help to rebase before starting to publish. So if we can run a release build first to cache the artifacts, we can then rebase and publish with a lower chance of something landing in master

Reply:
I'm not sure if that's possible unfortunately.

`cargo publish` runs a dev build against the newest possible dependencies from crates.io, and I can't find where it caches that build, if it caches it at all. If you keep running `cargo publish --dry-run` it'll redo the build each time from scratch

Reply:
Oh dang well then maybe we do a rebase right before pushing the commit?

Reply:
I guess that doesn't work either because we want the commit to be next to avoid confusing what was actually published?

Reply:
Yeah exactly, `cargo release` does bump + commit + publish + tag + push, and we should do that against the newest commit